File: Backend/src/seeders/createDefaultUsers.js
const sequelize = require('../config/database');
const User = require('../models/User');

const defaultUsers = [
  {
    email: 'admin@mathmaster.com',
    password: 'Admin123!',
    firstName: 'Administrador',
    lastName: 'Sistema',
    role: 'admin',
    isActive: true,
  },
  {
    email: 'docente@mathmaster.com',
    password: 'Docente123!',
    firstName: 'Profesor',
    lastName: 'Demo',
    role: 'teacher',
    isActive: true,
  },
  {
    email: 'estudiante@mathmaster.com',
    password: 'Estudiante123!',
    firstName: 'Estudiante',
    lastName: 'Demo',
    role: 'student',
    isActive: true,
  },
];

const seedUsers = async () => {
  try {
    // Sincronizar base de datos
    await sequelize.sync({ force: true });
    console.log('✅ Base de datos sincronizada');

    // Crear usuarios
    for (const userData of defaultUsers) {
      const user = await User.create(userData);
      console.log(`✅ Usuario creado: ${user.email} (${user.role})`);
    }

    console.log('\n🎉 Usuarios por defecto creados exitosamente!\n');
    console.log('='.repeat(60));
    console.log('📋 CREDENCIALES DE ACCESO:');
    console.log('='.repeat(60));
    console.log('\n👨‍💼 ADMINISTRADOR:');
    console.log('   Email: admin@mathmaster.com');
    console.log('   Contraseña: Admin123!');
    console.log('\n👩‍🏫 DOCENTE:');
    console.log('   Email: docente@mathmaster.com');
    console.log('   Contraseña: Docente123!');
    console.log('\n👨‍🎓 ESTUDIANTE:');
    console.log('   Email: estudiante@mathmaster.com');
    console.log('   Contraseña: Estudiante123!');
    console.log('\n' + '='.repeat(60));

    process.exit(0);
  } catch (error) {
    console.error('❌ Error al crear usuarios:', error);
    process.exit(1);
  }
};

seedUsers();

File: Backend/create_default_users.py
"""Script para crear usuarios por defecto"""
import sys
import os

# Agregar el directorio raíz al path
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))

from app.database import SessionLocal
from app.models import User, UserRole
from app.auth import get_password_hash


def create_default_users():
    """Crear usuarios por defecto"""
    db = SessionLocal()

    try:
        print("👥 Creando usuarios por defecto...")

        # Usuarios por defecto
        default_users = [
            {
                "email": "admin@mathmaster.com",
                "password": "admin123",
                "first_name": "Admin",
                "last_name": "Sistema",
                "role": UserRole.admin
            },
            {
                "email": "docente@mathmaster.com",
                "password": "docente123",
                "first_name": "Juan",
                "last_name": "Profesor",
                "role": UserRole.teacher
            },
            {
                "email": "estudiante@mathmaster.com",
                "password": "estudiante123",
                "first_name": "María",
                "last_name": "García",
                "role": UserRole.student
            }
        ]

        for user_data in default_users:
            # Verificar si el usuario ya existe
            existing_user = db.query(User).filter(User.email == user_data["email"]).first()

            if not existing_user:
                new_user = User(
                    email=user_data["email"],
                    password=get_password_hash(user_data["password"]),
                    first_name=user_data["first_name"],
                    last_name=user_data["last_name"],
                    role=user_data["role"],
                    is_active=True
                )
                db.add(new_user)
                db.commit()
                print(f"✅ Usuario creado: {user_data['email']}")
            else:
                print(f"ℹ️  Usuario ya existe: {user_data['email']}")

        print("\n✅ Usuarios por defecto creados exitosamente!")
        print("\n🔐 Credenciales:")
        print("   Admin: admin@mathmaster.com / admin123")
        print("   Docente: docente@mathmaster.com / docente123")
        print("   Estudiante: estudiante@mathmaster.com / estudiante123")

    except Exception as e:
        print(f"❌ Error: {e}")
        db.rollback()
        import traceback
        traceback.print_exc()
    finally:
        db.close()


if __name__ == "__main__":
    create_default_users()
